### ISSUE TITLE
time_local how to use with transform?

### ISSUE THREAD

**Issue by wazerstar** [caddyserver/transform-encoder#25]
Hey Since latest caddy 2.6.2 time_local was [added](https://github.com/caddyserver/caddy/pull/5108).

Currently I'm using transform in my logger like this, and would like to add local logging time with this, how is it done?

```
log {
        format transform [{ts}] - User={user_id} - X-Forwarded-For={request>headers>X-Forwarded-For} - remote_ip={request>remote_ip} Country={request>headers>Cf-Ipcountry} {request>method} {request>headers>X-Forwarded-Proto} {request>host} {request>uri} {request>headers>Referer>[0]} {request>headers>User-Agent>[0]} - {request>proto} {status} {size} - {
                time_format "02/Jan/2006 15:04:05 -0700"
        }

        output file path\file.log {
                roll true # Rotate logs, enabled by default
                roll_size_mb 5 # Set max size 5 MB
                roll_gzip true # Whether to compress rolled files
                #roll_local_time true # Use localhost time
                roll_keep 2 # Keep at most 2 log files
                roll_keep_days 7 # Keep log files for 7 days
        }
}
```
                
Adding time_local to it just spews out error        

There's also a new console format added.

https://github.com/caddyserver/caddy/pull/5109

**Comment by mohammed90** [comment]
> Adding time_local to it just spews out error

What's the error?

> There's also a new console format added.

It's irrelevant to this module. This module is dependent on the json format. It reads the json log line, and extracts values from it.

**Comment by francislavoie** [comment]
`time_local` should be added in here, I guess https://github.com/caddyserver/transform-encoder/blob/f79842b793d445b1935daaaf350f0357e72ddc7f/caddyfile.go#L63

**Comment by MrOggy85** [comment]
tl;dr
https://caddy.community/t/set-timezone-for-caddy-2-official-docker-image/8622

I forked this plugin and hardcoded `TimeLocal = True` (https://github.com/MrOggy85/transform-encoder) and then built my own caddy image. However, it had no effect. The reason being that caddy by default does not read `TZ` env var. 

I suggest closing this issue and mention this in the README.md of this plugin @mohammed90

**Comment by francislavoie** [comment]
Did you try `time_local`? We added it in a recent version of Caddy. https://caddyserver.com/docs/caddyfile/directives/log#format-modules

**Comment by MrOggy85** [comment]
@francislavoie I did this: https://github.com/caddyserver/transform-encoder/commit/7bc423a2e7892d62adbf05f8b71a39c67b26240d

Whatever option was chosen I always set TimeLocal to true. But, it did not have any effect. 

I have no doubt that Go can use the TZ env var, but if tzdata is not present in the OS, then how can it use it?

**Comment by francislavoie** [comment]
Yeah that's true, whatever environment you run in needs tzdata installed. I wasn't trying to imply that's not the case.

Also my previous comment was from before we added `time_local`, it was just an idea at the time.
